### ISSUE TITLE
Add individual volume level

### ISSUE THREAD

**Issue by mefromthepast** [bigbluebutton/bigbluebutton#9046]
The voice level between different participants is not equal at most times, and sometimes the microphones of individual users are especially loud or silent.

It would be nice if BBB would support the adjustment of volume locally (comparable to what the mumble desktop client offers).

**Comment by iSamof** [comment]
That would greatly enhance user experience, without having to constantly adjust one's own volume depending on who is speaking.

**Comment by grafnull** [comment]
This issue is the most asked question I get from users. So it would be really nice, if this would be implemented.

**Comment by mtsonline** [comment]
currently the audiostream is one mix of all the users audio, that is mixed in freeswitch before sending to participants.
so you would have to set the volume level before the mix and there would be only one global volume setting per room, if this could be implimented at all.

I would love to see sfu for audio (no mixing but sending all audio streams to all users) this would help to encrease audio quality (for smaller rooms and good bandwith)  and maybe would allow to change the volume per user in the htmlclient ...

**Comment by s1github** [comment]
Hi all, thank you for the great work.
Can we imagine a way for the moderator to adjust individual level volume at the freeswitch input, as described here : 
https://freeswitch.org/confluence/display/FREESWITCH/mod_dptools%3A+set+audio+level

**Comment by frank-miller9** [comment]
An auto-adjustment of volume would be really great. I think a manual one would be nice to have, but in real live really hard to use, because then the moderator can't concentrate on the meeting itself but is always busy adjusting volumes.

**Comment by 0xpr03** [comment]
+1 for this because it's just a hard requirement compared to jitsi, TS3, discord and mumble

**Comment by halbGefressen** [comment]
Maybe you could let each user adjust their volume themselves and then add a reference loudness value that is equal for all users so in the end every user is the same volume?

**Comment by AlEscher** [comment]
Even just an option to adjust the volume of the BBB room in general would be nice. For me the volume is always extremely loud, and the only option is to lower the System / Browser volume

**Comment by Hu1buerger** [comment]
Is this still not implemented?

**Comment by ffdixon** [comment]
Not yet -- there is more work planned for audio in BigBlueButton, but not a volume control for audio.  Still, this issue will remain open to keep this as an enhancement request.

**Comment by Freso** [comment]
One volume setting per room would be a vast improvement - or even just an overall master volume setting - would be a _massive_ improvement, since the BBB audio is completely overwhelming all other system audio, incl. audio from other browser tabs that may be playing simultaneously. Going into pulseaudio or whatever to adjust volume of the specific process is quite a hassle and will also often have to be redone across refreshes/reloads etc., where a site-volume setting can be stored in cookie or local storage.

**Comment by Hu1buerger** [comment]
@Freso i would disagree. Unfortunately i was forced to have to experience some spearkes being way to loud while others are nearly silent. They were in the same room.

In this case controlling the volume of the room is inadequate. This behaviour is not treatable by using the chrome extension volume master.

**Comment by nibah** [comment]
I would like to work on this in the next half year.

**Comment by 0xpr03** [comment]
> Unfortunately i was forced to have to experience some spearkes being way to loud while others are nearly silent. They were in the same room.

Compression would also help

**Comment by nibah** [comment]
Unfortunately my plans had to change, and I won't be able to work on this.

**Comment by davidsiecke** [comment]
For the record, this topic was also addressed at BigBlueButton World 2022 in the talk on usability and accessibility, and it was discussed afterwards whether it would be possible to implement this as a functionality for the presenter. You can revisit the discussion in the [session recording from minute 48:33 on](https://youtu.be/_H9sIk8UJnM?t=2913).

**Comment by tschammnut** [comment]
I came here looking for a way to adjust volume levels individually, but after reading the comments, I think I would actually prefer a "properly" normalized/compressed/limited mix from the server, as bandwidth issues arise so frequently already (from my experience, at least). If there was a solution to the problem of one speaker being extremely loud while others being almost inaudible that did not require me to intervene manually, I would definitely prefer that.

**Comment by aurorasmiles** [comment]
oh go away stalebot, no one likes you, this is still a valid feature request

**Comment by emoxam** [comment]
well, any news ?

**Comment by mittler-works** [comment]
The ticket is now old enough for elementary school and describes a feature I still really miss. Is there any sign of life indicating that this feature is still on the list?
